esp-ieee802154/src/lib.rs

```diff
@@ -210,9 +213,84 @@ impl Ieee802154 {
 
         Ok(())
     }
+
+    /// Transmit a raw frame
+    pub fn transmit_raw(&mut self, frame: &[u8]) -> Result<(), Error> {
+        self.transmit_buffer[1..][..frame.len()].copy_from_slice(frame);
+        self.transmit_buffer[0] = frame.len() as u8;
+
+        ieee802154_transmit(self.transmit_buffer.as_ptr() as *const u8, false); // what about CCA?
+
+        Ok(())
+    }
+
+    pub fn set_tx_done_callback(&mut self, callback: &mut (dyn FnMut() + Send)) {
+        critical_section::with(|cs| {
+            let mut tx_done_callback = TX_DONE_CALLBACK.borrow_ref_mut(cs);
+            tx_done_callback.replace(unsafe { core::mem::transmute(callback) });
+        });
+    }
+
+    pub fn clear_tx_done_callback(&mut self) {
+        critical_section::with(|cs| {
+            let mut tx_done_callback = TX_DONE_CALLBACK.borrow_ref_mut(cs);
+            tx_done_callback.take();
+        });
+    }
+
+    pub fn set_rx_available_callback(&mut self, callback: &mut (dyn FnMut() + Send)) {
```

Comment:
Hmm, some bad things will happen if we don't have some lifetimes here. If the closure is dropped the stored reference (which we transmute to 'static) will point to invalid memory, so when the callback is called ub will happen.

Reply:
Oh yes .... it should life at least as long as the Ieee802154 struct 👍 then it should get unset in drop